cmd/posix.go

```diff
@@ -1643,12 +1652,14 @@ func (s *posix) VerifyFile(volume, path string, fileSize int64, algo BitrotAlgor
 		}
 		n, err = file.Read(buf)
 		if err != nil {
+			// Read's failed for object with right size, at different offsets.
 			return err
 		}
 		size -= int64(n)
 		h.Write(buf)
 		if !bytes.Equal(h.Sum(nil), hashBuf) {
-			return HashMismatchError{hex.EncodeToString(hashBuf), hex.EncodeToString(h.Sum(nil))}
+			return fmt.Errorf("hashes do not match expected %s, got %s",
```

Comment:
I think we should wrap this error with VerifyFileError() so shouldHealObjectOnDisk() can detect if an object needs to be healed or not.

Reply:
Wrapping is already done in defer.